### PR TITLE
Added fetchRoles flag when getting subscriptions

### PR DESCRIPTION
**Pull request by tirthct**
**Changed**
- Added fetchRoles flag while getting subscriptions'
- Added presentation logic to group subscriptions by accountIDs

**Tested**
- Subscriptions roles should be included when fetchRoles flag is passed : tested

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +129,7 @@ The following flags are available for `get subscriptions`:
 * Get first subscription by its externalClusterID: `ocm support get subscriptions [externalClusterID] --first`
 * Get all subscriptions by OrganizationID and include subscriptions that have Status as 'Reserverd' `ocm support get subscriptions [organizationID] "Status='Reserved'"`
 * Get subscription by its ID and include its reserved resources: `ocm support get subscriptions [subscriptionID] --fetchReservedResources`
+* Get first subscription by its cluster ID and include its roles: `ocm support get subscriptions [clusterID] --first --fetchReservedResources`
```
Comment (cristianoveiga) [inline README.md:132]: did you mean fetchRoles?

File: pkg/role_binding/role_binding.go [hunk 2]
```diff
@@ -31,16 +31,26 @@ type AccountRoleBinding struct {
...
 	ApplicationRoleBinding  = "Application"
+
+	ManagedByUser = "User"
 )
 
 func CreateRoleBinding(accountID string, roleID string, roleType string, resourceID *string) (*v1.RoleBinding, error) {
 	newRoleBinding := v1.NewRoleBinding().AccountID(accountID).RoleID(roleID).Type(roleType)
 	if roleType == SubscriptionRoleBinding {
 		newRoleBinding.SubscriptionID(*resourceID)
+		//TODO : In case of creating a subscription role binding, managedBy is set to true in uhc-account-manager. Behavior is maintained here for consistency
```
Comment (cristianoveiga) [inline pkg/role_binding/role_binding.go:51]: What's the action item here?

File: pkg/role_binding/role_binding.go [hunk 3]
```diff
@@ -153,3 +172,24 @@ func PresentRoleBindings(roleBindings []*v1.RoleBinding) []AccountRoleBinding {
...
+func PresentSubscriptionRoleBindings(roleBindings []*v1.RoleBinding) []SubscriptionsRoleBinding {
+	var subscriptionRoleBindings []SubscriptionsRoleBinding
+	uniqueRoleBindingsMap := make(map[string][]string)
+	for _, roleBinding := range roleBindings {
+		if uniqueAccountRoleBindings, ok := uniqueRoleBindingsMap[roleBinding.Account().ID()]; ok {
+			uniqueAccountRoleBindings = append(uniqueAccountRoleBindings, roleBinding.Role().ID())
+			uniqueRoleBindingsMap[roleBinding.Account().ID()] = uniqueAccountRoleBindings
+		} else {
+			uniqueAccountRoleBindings := []string{roleBinding.Role().ID()}
+			uniqueRoleBindingsMap[roleBinding.Account().ID()] = uniqueAccountRoleBindings
```
Comment (cristianoveiga) [inline pkg/role_binding/role_binding.go:185]: As discussed, I am not sure if grouping by AccountID will be very useful. Let's take a look in a few examples from the stage/integration environment to determine the best grouping strategy.

**Comment by tirthct** [comment]
Closing the PR since a similar PR is created at https://github.com/openshift-online/ocm-support-cli/pull/24

**Pull request closed without merging**